osu.Game/Rulesets/Objects/Legacy/LegacyRulesetExtensions.cs

```diff
@@ -55,7 +55,7 @@ public static float CalculateScaleFromCircleSize(float circleSize, bool applyFud
             // It works out to under 1 game pixel and is generally not meaningful to gameplay, but is to replay playback accuracy.
             const float broken_gamefield_rounding_allowance = 1.00041f;
 
-            return (float)(1.0f - 0.7f * IBeatmapDifficultyInfo.DifficultyRange(circleSize)) / 2 * (applyFudge ? broken_gamefield_rounding_allowance : 1);
+            return (float)Math.Max(0.02, (1.0f - 0.7f * IBeatmapDifficultyInfo.DifficultyRange(circleSize)) / 2 * (applyFudge ? broken_gamefield_rounding_allowance : 1));
```

Comment:
FWIW I still think this should be gone from this diff and only the parser change should remain.

Reply:
I do think a check of this sort is useful to prevent the function from returning a result out of range, but sure. I've reverted it in the latest commit.